### PR TITLE
externalresource(engine): Integrate all parts related to GC (Part I)

### PR DESCRIPTION
**Pull request by liuzix**
### What problem does this PR solve?

Issue Number: ref #5513 

### What is changed and how it works?
- Implemented `LocalFileResourceController`
- Implemented `mockCluster` and some integration tests for GC.
- Minor adjustments of method names.
- Fixed minor bugs.

#### Tests 

 - Unit test
 - Integration test

### Release note 

```release-note
None.
```

### CODE REVIEW

File: engine/pkg/externalresource/resourcetypes/local_file.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+	"github.com/pingcap/tiflow/engine/client"
+	"github.com/pingcap/tiflow/engine/pb"
+	derrors "github.com/pingcap/tiflow/engine/pkg/errors"
+	resModel "github.com/pingcap/tiflow/engine/pkg/externalresource/resourcemeta/model"
+)
+
+// LocalFileResourceType defines operations specific to
+// the local file type.
+type LocalFileResourceType struct {
```
Comment (maxshuang) [inline engine/pkg/externalresource/resourcetypes/local_file.go:27]: This name sounds like a Type.

Comment (liuzix) [reply engine/pkg/externalresource/resourcetypes/local_file.go:27]: You are right. I will change it.

**Comment by maxshuang** [comment]
rest LGTM.

File: engine/pkg/externalresource/resourcetypes/interfaces.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+import (
+	"context"
+
+	resModel "github.com/pingcap/tiflow/engine/pkg/externalresource/resourcemeta/model"
+)
+
+// ResourceTypeController is an interface providing relevant operations related
+// to one resource type.
+type ResourceTypeController interface {
```
Comment (maxshuang) [inline engine/pkg/externalresource/resourcetypes/interfaces.go:24]: IMHO, why not the `ResourceTypeHandler` since it only has `GetHandler` interface?

**Review by maxshuang — APPROVED** [review]

File: engine/client/executor_client.go [hunk 3]
```diff
@@ -15,19 +15,31 @@ package client
...
+
+	"github.com/pingcap/tiflow/engine/pb"
+	"github.com/pingcap/tiflow/pkg/retry"
 )
 
 // ExecutorClient defines an interface that supports sending gRPC from server
 // master to executor.
 type ExecutorClient interface {
-	baseExecutorClient
+	// baseExecutorClient
```
[the remark this answers is not in the document]
Comment (liuzix) [reply engine/client/executor_client.go:30]: Yes. It seems `baseExecutorClient` should not be part of the public interface.

File: engine/pkg/externalresource/resourcetypes/interfaces.go (hunk 2 above)
Comment (liuzix) [reply engine/pkg/externalresource/resourcetypes/interfaces.go:24]: More methods are planned in future PRs.

**Review by amyangfei — APPROVED** [review]

**Pull request merged**